clients/src/main/java/org/apache/kafka/common/requests/UpdateMetadataRequest.java

```diff
@@ -261,14 +261,11 @@ public UpdateMetadataRequest(Struct struct) {
 
     @Override
     public AbstractRequestResponse getErrorResponse(int versionId, Throwable e) {
-        switch (versionId) {
-            case 0:
-            case 1:
-                return new UpdateMetadataResponse(Errors.forException(e).code());
-            default:
-                throw new IllegalArgumentException(String.format("Version %d is not valid. Valid versions for %s are 0 to %d",
-                        versionId, this.getClass().getSimpleName(), ProtoUtils.latestVersion(ApiKeys.UPDATE_METADATA_KEY.id)));
-        }
+        if (versionId <= 2)
```

Comment:
You are right. I did add the new response version but did not recall that I set the response version explicitly. Then I realized that the version is set automatically according to the request version.